apps/ecommerce/frontend/src/components/Common/ProductCard/ProductCardBadge/ProductCardBadge.tsx

```diff
@@ -0,0 +1,37 @@
+import { Badge, Box, EntityStatus } from '@contentful/f36-components';
+import { ExternalResource, ExternalResourceError } from 'types';
+import { styles } from './ProductCardBadge.styles';
+
+interface ProductCardBadgeProps {
+  showHeaderMenu?: boolean;
+  externalResourceError?: ExternalResourceError;
+  resource?: ExternalResource;
+}
+
+const WithStyleWrapper = (children: JSX.Element) => (
+  <Box data-test-id="badge-style-wrapper" className={styles.badge}>
+    {children}
+  </Box>
+)
+
+const ProductCardBadge = (props: ProductCardBadgeProps) => {
+  const { showHeaderMenu, externalResourceError, resource = {} } = props;
+  const { error, errorMessage, errorStatus } = externalResourceError || {};
+
+  // TO DO: Ensure the status/badge state is consistent based on data provided
+  const getDisplayStatus = () => {
+    if (resource.status) return resource.status;
+    return resource.availableForSale ? 'Available' : 'Not Available'
+  }
```

Comment:
perhaps this is just still in flux, but would love to align on the 'status' piece of a resource right now and how that renders via the badge.

Reply:
Perfect that sounds great!